src/webapp/components/import-export/ImportExport.tsx

```diff
@@ -0,0 +1,109 @@
+        setAnchorEl(event.currentTarget);
+    };
+
+    const closeMenu = () => {
+        setMenuOpen(false);
+    };
+
+    const orgUnitsField = settings.get("organisationUnitsField");
+    const { exportUsersToCSV, exportUsersToJSON, exportEmptyTemplate } = useExportUsers({
+        columns,
+        filterOptions,
+        orgUnitsField,
+        onSuccess: closeMenu,
+    });
+
+    const importFromFile = () => {
+        fileDialog({ accept: ["text/csv", "application/json"] })
+            .then((files: FileList) => {
+                setProcessing(true);
+                const file = files[0];
+                if (!file) return;
+                if (file.type === "text/csv") {
+                    return importFromCsv(d2, file, { maxUsers, orgUnitsField });
+                } else if (file.type === "application/json") {
+                    return importFromJson(d2, file, { maxUsers, orgUnitsField });
+                }
+            })
+            .then(onImport)
+            .catch(err => snackbar.error(err.toString()))
+            .finally(() => {
+                closeMenu();
+                setProcessing(false);
+            });
+    };
+
+    return (
+        <div className="data-table-import-export">
+            <IconButton onClick={openMenu} tooltipPosition="bottom-left" tooltip={i18n.t("Import/Export")}>
+                <ImportExportIcon />
+            </IconButton>
+
+            {isProcessing && <ModalLoadingMask />}
+
+            <Popover
+                open={isMenuOpen}
+                anchorEl={anchorEl}
+                anchorOrigin={{
+                    vertical: "center",
+                    horizontal: "center",
+                }}
+                transformOrigin={{
+                    vertical: "top",
+                    horizontal: "right",
+                }}
+                onClose={closeMenu}
+            >
+                <Menu>
+                    <MenuItem leftIcon={<ImportIcon />} onClick={importFromFile}>
+                        {i18n.t("Import")}
+                    </MenuItem>
+                    <MenuItem leftIcon={<ExportIcon />} onClick={exportUsersToCSV}>
+                        {i18n.t("Export to CSV")}
+                    </MenuItem>
+                    <MenuItem leftIcon={<ExportIcon />} onClick={exportUsersToJSON}>
+                        {i18n.t("Export to JSON")}
+                    </MenuItem>
+                    <MenuItem leftIcon={<ExportIcon />} onClick={exportEmptyTemplate}>
+                        {i18n.t("Export empty template")}
+                    </MenuItem>
+                </Menu>
+            </Popover>
+        </div>
+    );
+};
+
+export type FilterOption = { search: string; sorting: { field: string; order: "asc" | "desc" } };
+
+export type ImportExportProps = {
+    columns: ColumnMappingKeys[];
+    filterOptions: FilterOption;
+    onImport: (result: any) => void;
+    maxUsers: number;
+    settings: Record<string, any>;
```

Comment:
I think we can use the Settings class as a type, right? `src/legacy/models/settings.js`

```ts
class Settings {
 ...
}

export type ImportExportProps = {
    columns: ColumnMappingKeys[];
    filterOptions: FilterOption;
    onImport: (result: any) => void;
    maxUsers: number;
    settings: Settings;
};
```